### ISSUE TITLE
New functions

### ISSUE THREAD

**Issue by mrguy121** [zhuowei/MCPELauncher#881]
Add the following functions
ModPe.setBlockId(item Id to be changed to Id that's computable with set block)
Item.addGun(Id,damage,speed)
ModPe.addDimension(dimension id,block for portal,item for portal start,texture for portal,block for dimension to be made out of)
Player.addCreativeTab(item icon for picture,tab number)
Player.addItemCreativeInvTab(tab number)

**Comment by zedman2000** [comment]
ModPe.setBlockId(item Id to be changed to Id that's computable with set block)
-- just use setTile and replace the block

Item.addGun(Id,damage,speed)
--guns are not in Minecraft, use Desno365's mod

ModPe.addDimension(dimension id,block for portal,item for portal start,texture for portal,block for dimension to be made out of)
-- probably not going to happen due to the limitations of MCPE

Player.addCreativeTab(item icon for picture,tab number)
--I'll let @zhuowei comment on this one, although I'm guessing due to HOW the tabs are generated, probably not going to happen.

Player.addItemCreativeInvTab(tab number)
-- See above

**Comment by mrguy121** [comment]
For add gun I meant add item that shoots arrows
For the block Id thing I meant seeing as you only have limited block ids you can change an item I'd such as 1000 to work with the block.defineBlock command

**Comment by zedman2000** [comment]
@mrguy121 I'm not sure I follow regarding the block ID, are you saying you want to be able to place blocks above 256?  If so, that is a MCPE issue, unlike PC MCPE cannot do that.  You can use block IDS though...so for each blockID you can define 16 blocks.  I did that with a few of my mods, where I redefined a block so I did not use any additional IDS.  Then later on I just use block data to use my "new" blocks.

**Comment by mrguy121** [comment]
Yes

**Comment by zedman2000** [comment]
You can't do that.  it is a Pocket Edition Limitation.

**Comment by KenMizz** [comment]
addDimension is impossible

**Comment by WIPocket** [comment]
Also it has to start by ModPE

**Comment by DiegoGTRatty** [comment]
Wasn't the world format going to be re-coded as a part of the 0.15 update (after the "realms" update of course)? @zedman2000

**Comment by WIPocket** [comment]
No. They are going to make a more pc-like world generator, but the format stays. If you open an old world, it will work.
